examples/08_distributed_manager.py

```diff
@@ -0,0 +1,190 @@
+
+# %%
+import numpy as np
+import torch
+from dotenv import load_dotenv
+
+load_dotenv()  # TODO: make common example prep function
+from loguru import logger
+from modulus.distributed import DistributedManager
+
+DistributedManager.initialize()  # Only call this once in the entire script!
+dist = DistributedManager()
+assert (  # noqa: S101
+    dist._distributed
+), "Looks like torch distributed isn't set up. Check your env variables!"
+
+logger.info(
+    f"Inference runner {dist.rank} of {dist.world_size} with device {dist.device}"
+)
+# %%
+# Next the needed components get initialized.
+# Rigorous parallel support is not part of Earth2Studio's design goals, there are some
+# spots where potential race conditions can occur. Thus some additional care should be
+# taken to ensure safe parallel inference.
+
+# %%
+from earth2studio.data import ARCO
+from earth2studio.io import ZarrBackend
+from earth2studio.models.px import DLWP
+
+# Load model
+package = DLWP.load_default_package()
+if dist.rank == 0:
+    model = DLWP.load_model(package)
+
+torch.distributed.barrier()
+if dist.rank != 0:
+    model = DLWP.load_model(package)
+
+# %%
+# When loading models that are built into Earth2Studio, the model's checkpoint files
+# will be downloaded into the machines cache. If each inference process has access to
+# the same cache location, then only one should download the checkpoint triggered by
+# :py:func:`load_model`.
+#
+# Here :py:class:`earth2studio.models.px.DLWP` checkpoint files are first downloaded by
+# process 0 and then loaded by other processes.
+
+# %%
+
+# Create the data source
+data = ARCO()
+
+# %%
+# The remote date store will place cached data into separate caches for each process.
+# This makes the download of initial state data safe during parallel inference but also
+# means that multiple jobs will download the same date-time if needed.
+
+# %%
+chunks = {"time": 1}
+io = ZarrBackend(file_name=f"outputs/08_output_{dist.rank}.zarr", chunks=chunks)
+
+# %%
+# Earth2Studio does not provide distributed IO support. The recommendation is to always
+# output data for each process to a separate file, then aggregate the data during post
+# processing.
+#
+# Execute the Workflow
+# --------------------
+# Next we can run the workflow. This example will run inference for a random date across
+# several years and just save total column water vapor. Shard the initial date-times
+# across the each process. The distributed manager will provide the device ID for the
+# process.
+
+# %%
+import earth2studio.run as run
+
+times = np.array([f"200{i:d}-06-01T00:00:00" for i in range(0, 6)])
+assert (  # noqa: S101
+    len(times) > dist.world_size
+), "Inference runs should be greater than processes"
+time_shard = np.array_split(times, dist.world_size)[dist.rank]
+
+nsteps = 20
+output_coords = {"variable": np.array(["tcwv"])}
+io = run.deterministic(
+    time_shard, nsteps, model, data, io, output_coords, device=dist.device
```

Comment:
maybe just ensure that `output_coords = output_coords` for safety.

Reply:
Added